core/src/main/java/discord4j/core/event/dispatch/ChannelDispatchHandlers.java

```diff
@@ -234,95 +89,37 @@
         return Mono.just(new PinsUpdateEvent(context.getGateway(), context.getShardInfo(), channelId, guildId, timestamp));
     }
 
-    static Mono<? extends Event> channelUpdate(DispatchContext<ChannelUpdate> context) {
+    static Mono<? extends Event> channelUpdate(DispatchContext<ChannelUpdate, ChannelData> context) {
         Channel.Type type = Channel.Type.of(context.getDispatch().channel().type());
-
-        switch (type) {
-            case GUILD_TEXT: return textChannelUpdate(context);
-            case DM:
-                throw new UnsupportedOperationException("Received channel_update for a DM on a bot account!");
-            case GUILD_VOICE: return voiceChannelUpdate(context);
-            case GROUP_DM:
-                throw new UnsupportedOperationException("Received channel_update for a group on a bot account!");
-            case GUILD_CATEGORY: return categoryUpdate(context);
-            case GUILD_NEWS: return newsChannelUpdate(context);
-            case GUILD_STORE: return storeChannelUpdate(context);
-            default: throw new AssertionError();
-        }
-    }
-
-    private static Mono<TextChannelUpdateEvent> textChannelUpdate(DispatchContext<ChannelUpdate> context) {
         GatewayDiscordClient gateway = context.getGateway();
         ChannelData channel = context.getDispatch().channel();
         GuildMessageChannel current = getConvertibleChannel(gateway, channel);
```

Comment:
This should be removed as it is unused and causes an error if the channel type is not supported

Reply:
removed